### PR TITLE
Made FolderObserver a configured component instead of a ManagedService

### LINKED ISSUE #3562: ScriptServiceUtil not initialized yet on startup
```
2017-05-31 20:09:57.118 [ERROR] [xbase.resource.BatchLinkableResource] - resolution of uriFragment '|::0.2.1.2.0.1.2.2.2.0.6.6.6::0::/1' failed.
java.lang.IllegalStateException: ScriptServiceUtil not initialized yet!
        at org.eclipse.smarthome.model.script.ScriptServiceUtil.getInstance(ScriptServiceUtil.java:88)
        at org.eclipse.smarthome.model.script.ScriptServiceUtil.getActionServices(ScriptServiceUtil.java:114)
        at org.eclipse.smarthome.model.script.internal.engine.ServiceTrackerActionServiceProvider.get(ServiceTrackerActionServiceProvider.java:21)
        at org.eclipse.smarthome.model.script.internal.engine.ServiceTrackerActionServiceProvider.get(ServiceTrackerActionServiceProvider.java:1)
        at org.eclipse.smarthome.model.script.scoping.ScriptImplicitlyImportedTypes.getActionClasses(ScriptImplicitlyImportedTypes.java:100)
        at org.eclipse.smarthome.model.script.scoping.ScriptImplicitlyImportedTypes.getStaticImportClasses(ScriptImplicitlyImportedTypes.java:94)
        at org.eclipse.xtext.xbase.scoping.batch.ImplicitlyImportedFeatures.getStaticImportClasses(ImplicitlyImportedFeatures.java:66)
        at org.eclipse.xtext.xbase.scoping.batch.XbaseBatchScopeProvider.newSession(XbaseBatchScopeProvider.java:112)
        at org.eclipse.xtext.xbase.typesystem.internal.DefaultReentrantTypeResolver.resolve(DefaultReentrantTypeResolver.java:162)
        at org.eclipse.xtext.xbase.typesystem.internal.DefaultReentrantTypeResolver.reentrantResolve(DefaultReentrantTypeResolver.java:139)
        at org.eclipse.xtext.xbase.typesystem.internal.CachingBatchTypeResolver$LazyResolvedTypes.resolveTypes(CachingBatchTypeResolver.java:80)
        at org.eclipse.xtext.xbase.typesystem.internal.CachingBatchTypeResolver$2.process(CachingBatchTypeResolver.java:57)
        at org.eclipse.xtext.xbase.typesystem.internal.CachingBatchTypeResolver$2.process(CachingBatchTypeResolver.java:1)
        at org.eclipse.xtext.util.concurrent.IUnitOfWork$Void.exec(IUnitOfWork.java:37)
        at org.eclipse.xtext.util.OnChangeEvictingCache.execWithoutCacheClear(OnChangeEvictingCache.java:129)
        at org.eclipse.xtext.xbase.typesystem.internal.CachingBatchTypeResolver.doResolveTypes(CachingBatchTypeResolver.java:53)
        at org.eclipse.xtext.xbase.typesystem.internal.AbstractBatchTypeResolver.resolveTypes(AbstractBatchTypeResolver.java:69)
        at org.eclipse.xtext.xbase.resource.BatchLinkingService.resolveBatched(BatchLinkingService.java:60)
        at org.eclipse.xtext.xbase.resource.BatchLinkingService.resolveBatched(BatchLinkingService.java:41)
        at org.eclipse.xtext.xbase.resource.BatchLinkableResource.getEObject(BatchLinkableResource.java:117)
        at org.eclipse.emf.ecore.resource.impl.ResourceSetImpl.getEObject(ResourceSetImpl.java:223)
        at org.eclipse.emf.ecore.util.EcoreUtil.resolve(EcoreUtil.java:203)
        at org.eclipse.emf.ecore.util.EcoreUtil.resolve(EcoreUtil.java:259)
        at org.eclipse.emf.ecore.impl.BasicEObjectImpl.eResolveProxy(BasicEObjectImpl.java:1477)
        at org.eclipse.xtext.xbase.impl.XAbstractFeatureCallImplCustom.getFeature(XAbstractFeatureCallImplCustom.java:48)
        at org.eclipse.xtext.xbase.validation.XbaseValidator.checkDelegateConstructorIsFirst(XbaseValidator.java:927)
        at sun.reflect.GeneratedMethodAccessor13.invoke(Unknown Source)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)[:1.8.0_131]
        at java.lang.reflect.Method.invoke(Method.java:498)[:1.8.0_131]
        at org.eclipse.xtext.validation.AbstractDeclarativeValidator$MethodWrapper.invoke(AbstractDeclarativeValidator.java:118)[142:org.eclipse.xtext:2.9.2]
        at org.eclipse.xtext.validation.AbstractDeclarativeValidator.internalValidate(AbstractDeclarativeValidator.java:312)[142:org.eclipse.xtext:2.9.2]
        at org.eclipse.xtext.validation.AbstractInjectableValidator.validate(AbstractInjectableValidator.java:71)[142:org.eclipse.xtext:2.9.2]
        at org.eclipse.xtext.validation.CompositeEValidator.validate(CompositeEValidator.java:151)[142:org.eclipse.xtext:2.9.2]
        at org.eclipse.emf.ecore.util.Diagnostician.doValidate(Diagnostician.java:171)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:158)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:137)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.doValidateContents(Diagnostician.java:181)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:161)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:137)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.doValidateContents(Diagnostician.java:181)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:161)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:137)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.doValidateContents(Diagnostician.java:181)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:161)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:137)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.doValidateContents(Diagnostician.java:181)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:161)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:137)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.doValidateContents(Diagnostician.java:185)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:161)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:137)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.doValidateContents(Diagnostician.java:185)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:161)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:137)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.doValidateContents(Diagnostician.java:185)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:161)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:137)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.doValidateContents(Diagnostician.java:185)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:161)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:137)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.doValidateContents(Diagnostician.java:185)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:161)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:137)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.doValidateContents(Diagnostician.java:185)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:161)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:137)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.emf.ecore.util.Diagnostician.validate(Diagnostician.java:108)[65:org.eclipse.emf.ecore:2.11.1.v20150805-0538]
        at org.eclipse.smarthome.model.core.internal.ModelRepositoryImpl.validateModel(ModelRepositoryImpl.java:265)[123:org.eclipse.smarthome.model.core:0.9.0.201705301617]
        at org.eclipse.smarthome.model.core.internal.ModelRepositoryImpl.addOrRefreshModel(ModelRepositoryImpl.java:90)[123:org.eclipse.smarthome.model.core:0.9.0.201705301617]
        at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.checkFile(FolderObserver.java:244)[123:org.eclipse.smarthome.model.core:0.9.0.201705301617]
        at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.notifyUpdateToModelRepo(FolderObserver.java:170)[123:org.eclipse.smarthome.model.core:0.9.0.201705301617]
        at org.eclipse.smarthome.model.core.internal.folder.FolderObserver.updated(FolderObserver.java:148)[123:org.eclipse.smarthome.model.core:0.9.0.201705301617]
        at org.apache.felix.cm.impl.helper.ManagedServiceTracker.updated(ManagedServiceTracker.java:189)[3:org.apache.felix.configadmin:1.8.12]
        at org.apache.felix.cm.impl.helper.ManagedServiceTracker.updateService(ManagedServiceTracker.java:152)[3:org.apache.felix.configadmin:1.8.12]
        at org.apache.felix.cm.impl.helper.ManagedServiceTracker.provideConfiguration(ManagedServiceTracker.java:85)[3:org.apache.felix.configadmin:1.8.12]
        at org.apache.felix.cm.impl.ConfigurationManager$ManagedServiceUpdate.provide(ConfigurationManager.java:1461)[3:org.apache.felix.configadmin:1.8.12]
        at org.apache.felix.cm.impl.ConfigurationManager$ManagedServiceUpdate.run(ConfigurationManager.java:1417)[3:org.apache.felix.configadmin:1.8.12]
        at org.apache.felix.cm.impl.UpdateThread.run0(UpdateThread.java:141)[3:org.apache.felix.configadmin:1.8.12]
        at org.apache.felix.cm.impl.UpdateThread.run(UpdateThread.java:109)[3:org.apache.felix.configadmin:1.8.12]
        at java.lang.Thread.run(Thread.java:748)[:1.8.0_131]

```

See https://community.openhab.org/t/snapshot-930-not-starting-properly/29252

### PR DESCRIPTION
**Pull request by kaikreuzer**
The FolderObserver had some potential issues:
- The activate method missed to call `super.activate()` (how did that actually then ever work?)
- The list of ignoredFiles was not cleared on deactivation; so when deactivating and activating the component, it was possible that it was still filled.
- The addModelParser() method directly executed code that tried to parse ignoredFiles. BUT this method was potentially called BEFORE the component was activated, which can lead to `java.lang.IllegalStateException: ScriptServiceUtil not initialized yet!` exceptions.
- The component was a ManagedService, which received its configuration only after being activated and then manually called deactivate/activate (woooaaah, looks dangerous 😲 ).

This PR includes:
- I have now changed the implementation to receive the (mandatory) configuration directly during activation, so the service does not implement `ManagedService` anymore.
- The ignoredFiles as well as other fields are cleared upon deactivation.
- `super.activate()` is correctly called.
- All models are removed from the model repo upon component deactivation.
- I removed the dynamic configuration change possibility as anyhow the list of model folders is static and there isn't a real use case where someone would want to change this list during runtime. And if it happens, DS will simply deactivate the component and re-activate it again with a new config.

I have some hope that this can actually fix #3562. @sjka should definitely have a look over it!

Signed-off-by: Kai Kreuzer <kai@openhab.org>

### CODE REVIEW

File: bundles/model/org.eclipse.smarthome.model.core/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -25,6 +25,7 @@ Import-Package: com.google.common.base,
  org.eclipse.xtext.resource.impl,
  org.osgi.framework,
  org.osgi.service.cm,
```
Comment (sjka) [inline bundles/model/org.eclipse.smarthome.model.core/META-INF/MANIFEST.MF]: we don't need the cm package anymore, do we?

**Comment by sjka** [comment]
Thanks! 
What I missed so far was indeed the fact that the problematic calls came in via `FolderObserver.addModelParser(..)`, so this really looks like it could be the yet missing piece we were searching for.

**Comment by sjka** [comment]
However:

```
Failed tests: 
  FolderObserverTest.setUp:88->setUpServices:117 Folder Observer service cannot be found
Expected: is not null
     but: was null
  FolderObserverTest.setUp:88->setUpServices:117 Folder Observer service cannot be found
Expected: is not null
     but: was null
  FolderObserverTest.setUp:88->setUpServices:117 Folder Observer service cannot be found
Expected: is not null
     but: was null
  FolderObserverTest.setUp:88->setUpServices:117 Folder Observer service cannot be found
Expected: is not null
     but: was null
  FolderObserverTest.setUp:88->setUpServices:117 Folder Observer service cannot be found
Expected: is not null
     but: was null
  FolderObserverTest.setUp:88->setUpServices:117 Folder Observer service cannot be found
Expected: is not null
     but: was null
  FolderObserverTest.setUp:88->setUpServices:117 Folder Observer service cannot be found
Expected: is not null
     but: was null
  FolderObserverTest.setUp:88->setUpServices:117 Folder Observer service cannot be found
Expected: is not null
     but: was null
  FolderObserverTest.setUp:88->setUpServices:117 Folder Observer service cannot be found
Expected: is not null
     but: was null
  FolderObserverTest.setUp:88->setUpServices:117 Folder Observer service cannot be found
Expected: is not null
     but: was null
```

Looks like the service now is not running anmore in the tests because the required configuration is missing.
I wonder why this needs to be run as an OSGi test at all...

File: bundles/model/org.eclipse.smarthome.model.core/META-INF/MANIFEST.MF (hunk 1 above)
Comment (kaikreuzer) [inline bundles/model/org.eclipse.smarthome.model.core/META-INF/MANIFEST.MF]: correct, updated PR.

**Comment by kaikreuzer** [comment]
Ok, I'll deal with the tests (didn't have them in my workspace, so I missed them...)

**Comment by sjka** [comment]
I took the freedom to adapt the tests and pushed them here.
They got a little makeover while I ported them to Java, so that now they don't depend on the OSGi runtime anymore.

**Comment by kaikreuzer** [comment]
> Has been this. added in front of members by intention (also if it is not necessary)?

Yeah, it was probably done rather out of a mood. I added it while refactoring, because there were two similarly named variables, one being a field, the other being a local method variable. But I agree that we do not do it in general and I could have removed it here after the code was clean again anyhow.

**Comment by sjka** [comment]
> Should we create a parent class for JavaOSGiTest named JavaTest (some time -- not in this PR) so non-OSGi based test could use that base class?

Yes and no. It makes sense to have it - but I'm not so fond of base classes. How about a utility class and/or junit rule (depending on the purpose)?

**Pull request merged**